packages/react-components/src/shared/src/slots.tsx

```diff
@@ -22,7 +34,8 @@ export function getSlots(children, { _ = {}, ...slots }) {
 
         Children.forEach(children, x => {
             if (!isNil(x)) {
-                const slot = (x.props && x.props["slot"]) ?? (x.type && x.type[SLOT_KEY]);
+                const reactComponent = x as ReactElement;
```

Comment:
instead of doing this, could you do:

`Children.forEach(children, x: ReactElement => {`